handle checksum options (#127)
matrix-org/synapse-s3-storage-provider: README.md

    # to S3. Each thread manages a single connection. Default is 40.
    #
    #threadpool_size: 20

>     # Set request_checksum_calculation or response_checksum_validation

Reviewer: Please add some words about *why* a sysadmin may want to set these options (i.e. link to AWS' docs that explain what "when_supported" and "when_required" mean).

Author: Will do!

    # Default is None
    # request_checksum_calculation: "when_supported" | "when_required"

>     # response_checksum_validation: "when_supported" | "when_required"

Reviewer: Is there are reason not to make `"when_required"` the default, or will that break some known use cases?

    #
    #threadpool_size: 20

    # Set request_checksum_calculation or response_checksum_validation. 

>     # Depending on your S3 provider you may need to set these values. Esp. in case you are using a self-hosted system that does not support the functionality provided by AWS. 

Reviewer: ```suggestion
    # Depending on your S3 provider you may need to set these values,
    # especially if you are using a self-hosted system that does not
    # support the functionality provided by AWS.
```
